mp4parse_capi/src/lib.rs

```diff
@@ -698,17 +698,22 @@ fn create_sample_table(track: &Track) -> Option<Vec<mp4parse_indice>> {
     // Create sample table and calculate sample offset.
     let mut sample_table = Vec::new();
     let mut sample_size_iter = stsz.sample_sizes.iter();
-    let mut start_chunk = 0;
-    let mut end_chunk = 0;
-    let chunk_iter = stsc.samples.iter()
-        .map(|sample_chunk| {
-                // Convert the compact table to chunk range and samples per trunk.
-                start_chunk = end_chunk;
-                end_chunk += sample_chunk.first_chunk;
-                ((start_chunk .. end_chunk), sample_chunk.samples_per_chunk)
-            });
```

Comment:
We should calculate in reverse order so we can convert a the compact table to correct chunk range.